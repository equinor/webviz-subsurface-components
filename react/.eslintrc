{
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module"
  },
  "extends":  [
    "plugin:react/recommended",
    "eslint:recommended",
    "plugin:prettier/recommended"
  ],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx"],
      "plugins": ["@typescript-eslint"],
      "extends": [
        "plugin:react/recommended",
        "plugin:@typescript-eslint/recommended",
        "plugin:prettier/recommended"
      ]
    }
  ],
  "settings": {
    "react": {
      "version": "detect"
    }
  },
  "env": {
    "amd": true,
    "browser": true,
    "es6": true,
    "jasmine": true,
    "node": true
  }
}
