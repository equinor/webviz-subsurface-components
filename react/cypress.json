{
    "baseUrl": "http://localhost:6006/?path=/story/deckglmap--default",
    "video": true
}
