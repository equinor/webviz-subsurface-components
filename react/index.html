<html>
    <head>
        <title>Demo application of webviz-subsurface-components</title>
    </head>
    <body>
        <div id='root'></div>
        <link rel="stylesheet" type="text/css" href="output.css">
        <script src="output.js"></script>
    </body>
</html>
