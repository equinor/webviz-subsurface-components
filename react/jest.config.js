/**
 * The jest tests had an error: SyntaxError: Cannot use import statement outside a module
 * After some investigation, it leads to this page https://github.com/facebook/jest/issues/9395.
 */
module.exports = {
    moduleNameMapper: {
        "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$":
            "<rootDir>/__mocks__/fileMock.js",
        "\\.(css|less)$": "<rootDir>/__mocks__/styleMock.js",
    },
    transform: {
        "\\.(js|ts|jsx|tsx)$": [
            "babel-jest",
            { configFile: "./config/babel.config.json" },
        ],
    },
    transformIgnorePatterns: ["<rootDir>/node_modules/(?!(@webviz)/)"],
};
