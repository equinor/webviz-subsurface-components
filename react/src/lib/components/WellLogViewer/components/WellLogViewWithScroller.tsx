import React, { Component, ReactNode } from "react";

import { LogViewer } from "@equinor/videx-wellog";
import { TrackMouseEvent } from "./WellLogView";

import WellLogView from "./WellLogView";

import { WellLog } from "./WellLogTypes";
import { Template } from "./WellLogTemplateTypes";
import { ColorTable } from "./ColorTableTypes";

import { WellLogController } from "./WellLogView";

import Scroller from "./Scroller";

interface Props {
    welllog: WellLog;
    template: Template;
    colorTables: ColorTable[];
    horizontal?: boolean;
    primaryAxis: string;

    hideTitles?: boolean;
    hideLegend?: boolean;

    axisTitles: Record<string, string>;
    axisMnemos: Record<string, string[]>;

    maxVisibleTrackNum?: number; // default is horizontal ? 3: 5
    maxContentZoom?: number; // default is 256

    checkDatafileSchema?: boolean;

    // callbacks:
    onCreateController?: (controller: WellLogController) => void;
    onInfo?: (
        x: number,
        logController: LogViewer,
        iFrom: number,
        iTo: number
    ) => void;

    onTrackScroll?: () => void; // called when track scrolling are changed
    onTrackSelection?: () => void; // called when trackselection is changed
    onContentRescale?: () => void; // called when content zoom and scrolling are changed
    onContentSelection?: () => void; // called when content zoom and scrolling are changed

    onTrackMouseEvent?: (wellLogView: WellLogView, ev: TrackMouseEvent) => void; // called when mouse click on a track
    onTemplateChanged?: () => void; // called when track scrolling are changed
}

class WellLogViewWithScroller extends Component<Props> {
    controller: WellLogController | null;
    scroller: Scroller | null;

    constructor(props: Props) {
        super(props);

        this.controller = null;
        this.scroller = null;

        this.onCreateController = this.onCreateController.bind(this);

        this.onScrollerScroll = this.onScrollerScroll.bind(this);
        this.onTrackScroll = this.onTrackScroll.bind(this);
        this.onTrackSelection = this.onTrackSelection.bind(this);

        this.onContentRescale = this.onContentRescale.bind(this);
        this.onContentSelection = this.onContentSelection.bind(this);
    }

    componentDidMount(): void {
        this.setScrollerPosAndZoom();
    }

    shouldComponentUpdate(nextProps: Props): boolean {
        return !Object.is(this.props, nextProps);
    }

    updateReadoutPanel(): void {
        const controller = this.controller;
        if (controller)
            controller.selectContent(controller.getContentSelection()); // force to update readout panel
    }

    // callback function from WellLogView
    onCreateController(controller: WellLogController): void {
        this.controller = controller;
        if (this.props.onCreateController)
            // set callback to component's caller
            this.props.onCreateController(controller);
    }
    // callback function from WellLogView
    onTrackScroll(): void {
        this.setScrollerPosAndZoom();
        if (this.props.onTrackScroll) this.props.onTrackScroll();
    }
    // callback function from WellLogView
    onTrackSelection(): void {
        if (this.props.onTrackSelection) this.props.onTrackSelection();
    }
    // callback function from WellLogView
    onContentRescale(): void {
        this.setScrollerPosAndZoom();
        if (this.props.onContentRescale) this.props.onContentRescale();
    }
    // callback function from WellLogView
    onContentSelection(): void {
        if (this.props.onContentSelection) this.props.onContentSelection();
    }

    // callback function from Scroller
    onScrollerScroll(x: number, y: number): void {
        const controller = this.controller;
        if (controller) {
            const fContent = this.props.horizontal ? x : y; // fraction
            controller.scrollContentTo(fContent);

            const posTrack = this.calcPosTrack(this.props.horizontal ? y : x);
            controller.scrollTrackTo(posTrack);
        }
    }

    calcPosTrack(f: number): number {
        const controller = this.controller;
        if (!controller) return 0;
        const posMax = controller.getTrackScrollPosMax();
        const posTrack = f * posMax;
        return Math.round(posTrack);
    }

    getContentPosFraction(): number {
        const controller = this.controller;
        if (!controller) return 0;
        const baseDomain = controller.getContentBaseDomain();
        const domain = controller.getContentDomain();
        const w = baseDomain[1] - baseDomain[0] - (domain[1] - domain[0]);
        return w ? (domain[0] - baseDomain[0]) / w : 0;
    }
    getTrackPosFraction(): number {
        const controller = this.controller;
        if (!controller) return 0;
        return controller.getTrackScrollPosMax()
            ? controller.getTrackScrollPos() / controller.getTrackScrollPosMax()
            : 0.0;
    }

    setScrollerPosAndZoom(): void {
        let x: number, y: number;
        let xZoom: number, yZoom: number;
        const scroller = this.scroller;
        if (!scroller) return;
        const controller = this.controller;
        if (!controller) {
            x = y = 0.0;
            xZoom = yZoom = 1.0;
        } else {
            const contentZoom = controller.getContentZoom();
            const trackZoom = controller.getTrackZoom();
            xZoom = this.props.horizontal ? contentZoom : trackZoom;
            yZoom = this.props.horizontal ? trackZoom : contentZoom;

            const fContent = this.getContentPosFraction();
            const fTrack = this.getTrackPosFraction();
            x = this.props.horizontal ? fContent : fTrack;
            y = this.props.horizontal ? fTrack : fContent;
        }

        scroller.zoom(xZoom, yZoom);

        let shouldUpdateScroller = 2;
        {
            // compare with current values from scroller
            const _x = scroller.getScrollX();
            const _y = scroller.getScrollY();
            const _posTrack = this.calcPosTrack(
                this.props.horizontal ? _y : _x
            );
            const posTrack = this.calcPosTrack(this.props.horizontal ? y : x);
            if (posTrack === _posTrack) {
                shouldUpdateScroller--;
                this.props.horizontal ? (y = _y) : (x = _x);
            }
            const _fContent = this.props.horizontal ? _x : _y;
            const fContent = this.props.horizontal ? x : y;
            if (Math.abs(fContent - _fContent) < 0.001) {
                shouldUpdateScroller--;
                this.props.horizontal ? (x = _x) : (y = _y);
            }
        }
        if (shouldUpdateScroller) scroller.scrollTo(x, y);
    }

    render(): ReactNode {
        return (
            <Scroller
                ref={(el) => (this.scroller = el as Scroller)}
                onScroll={this.onScrollerScroll}
            >
                <WellLogView
                    welllog={this.props.welllog}
                    template={this.props.template}
                    colorTables={this.props.colorTables}
                    horizontal={this.props.horizontal}
                    hideTitles={this.props.hideTitles}
                    hideLegend={this.props.hideLegend}
                    maxVisibleTrackNum={this.props.maxVisibleTrackNum}
                    maxContentZoom={this.props.maxContentZoom}
                    checkDatafileSchema={this.props.checkDatafileSchema}
                    primaryAxis={this.props.primaryAxis}
                    axisTitles={this.props.axisTitles}
                    axisMnemos={this.props.axisMnemos}
                    onInfo={this.props.onInfo}
                    onCreateController={this.onCreateController}
                    onTrackMouseEvent={this.props.onTrackMouseEvent}
                    onTemplateChanged={this.props.onTemplateChanged}
                    onTrackScroll={this.onTrackScroll}
                    onTrackSelection={this.onTrackSelection}
                    onContentRescale={this.onContentRescale}
                    onContentSelection={this.onContentSelection}
                />
            </Scroller>
        );
    }
}

export default WellLogViewWithScroller;
