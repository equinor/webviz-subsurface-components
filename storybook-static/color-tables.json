[
    {
        "name": "Physics",
        "discrete": false,
        "description": "Full options color table",
        "colorNaN": [255, 255, 255],
        "colorBelow": [255, 0.0, 0.0],
        "colorAbove": [0.0, 0.0, 255],
        "colors": [
            [0.0, 255, 0.0, 0.0],
            [0.25, 182, 182, 0.0],
            [0.5, 0.0, 255, 0.0],
            [0.75, 0.0, 182, 182],
            [1.0, 0.0, 0.0, 255]
        ]
    },
    {
        "name": "Physics reverse",
        "discrete": false,
        "colors": [
            [0.0, 0.0, 0.0, 255],
            [0.25, 0.0, 182, 182],
            [0.5, 0.0, 255, 0.0],
            [0.75, 182, 182, 0.0],
            [1.0, 255, 0.0, 0.0]
        ]
    },
    {
        "name": "Rainbow",
        "discrete": false,
        "colors": [
            [0.0, 255, 0.0, 0.0],
            [0.2, 182, 182, 0.0],
            [0.4, 0.0, 255, 0.0],
            [0.6, 0.0, 182, 182],
            [0.8, 0.0, 0.0, 255],
            [1.0, 182, 0.0, 182]
        ]
    },
    {
        "name": "Rainbow reverse",
        "discrete": false,
        "colors": [
            [0.0, 182, 0.0, 182],
            [0.2, 0.0, 0.0, 255],
            [0.4, 0.0, 182, 182],
            [0.6, 0.0, 255, 0.0],
            [0.8, 182, 182, 0.0],
            [1.0, 255, 0.0, 0.0]
        ]
    },
    {
        "name": "Colors_set_1",
        "discrete": "true",
        "colors": [
            [0, 255, 13, 186],
            [1, 255, 64, 53],
            [2, 247, 255, 164],
            [3, 112, 255, 97],
            [4, 9, 254, 133],
            [5, 254, 4, 135],
            [6, 255, 5, 94],
            [7, 32, 50, 255],
            [8, 109, 255, 32],
            [9, 254, 146, 92],
            [10, 185, 116, 255],
            [11, 255, 144, 1],
            [12, 157, 32, 255],
            [13, 255, 26, 202],
            [14, 73, 255, 35]
        ]
    },
    {
        "name": "Stratigraphy",
        "discrete": true,
        "colorNaN" : [255, 64, 64],
        "colors": [
            [0, 255, 193, 0],
            [1, 255, 120, 61],
            [2, 255, 155, 76],
            [3, 255, 223, 161],
            [4, 226, 44, 118],
            [5, 255, 243, 53],
            [6, 255, 212, 179],
            [7, 255, 155, 23],
            [8, 255, 246, 117],
            [9, 255, 241, 0],
            [10, 255, 211, 178],
            [11, 255, 173, 128],
            [12, 248, 152, 0],
            [13, 154, 89, 24],
            [14, 0, 138, 185],
            [15, 82, 161, 40],
            [16, 219, 228, 163],
            [17, 0, 119, 64],
            [18, 0, 110, 172],
            [19, 116, 190, 230],
            [20, 0, 155, 212],
            [21, 0, 117, 190],
            [22, 143, 40, 112],
            [23, 220, 153, 190],
            [24, 226, 44, 118],
            [25, 126, 40, 111],
            [26, 73, 69, 43],
            [27, 203, 63, 42],
            [28, 255, 198, 190],
            [29, 135, 49, 45],
            [30, 150, 136, 120],
            [31, 198, 182, 175],
            [32, 166, 154, 145],
            [33, 191, 88, 22],
            [34, 255, 212, 179],
            [35, 251, 139, 105],
            [36, 154, 89, 24],
            [37, 186, 222, 200],
            [38, 0, 124, 140],
            [39, 87, 84, 83]
        ]
    },
    {
        "name": "Colors_set_3",
        "discrete": "true",
        "colors": [
            [0, 120, 181, 255],
            [1, 255, 29, 102],
            [2, 247, 255, 173],
            [3, 239, 157, 255],
            [4, 186, 255, 236],
            [5, 46, 255, 121],
            [6, 212, 255, 144],
            [7, 165, 255, 143],
            [8, 122, 255, 89],
            [9, 255, 212, 213]
        ]
    },
    {
        "name": "Porosity",
        "discrete": false,
        "colors": [
            [0.00, 255, 246, 117],
            [0.11, 255, 243, 53],
            [0.18, 255, 241, 0],
            [0.25, 155, 193, 0],
            [0.32, 255, 155, 23],
            [0.39, 255, 162, 61],
            [0.46, 255, 126, 45],
            [0.53, 227, 112, 24],
            [0.60, 246, 96, 31],
            [0.67, 229, 39, 48],
            [0.74, 252, 177, 170],
            [0.81, 236, 103, 146],
            [0.88, 226, 44, 118],
            [1.00, 126, 40, 111]
        ]
    }
]
