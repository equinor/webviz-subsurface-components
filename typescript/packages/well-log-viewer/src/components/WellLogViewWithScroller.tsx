import React, { useRef, useCallback, useEffect } from "react";

import WellLogView from "./WellLogView";
import type { WellLogViewProps } from "./WellLogView";
import { argTypesWellLogViewProp } from "./WellLogView";
import { _propTypesWellLogView } from "./WellLogView";

import type { WellLogController } from "./WellLogView";

import Scroller from "./Scroller";
import type { ScrollerRef } from "./Scroller";

export type WellLogViewWithScrollerProps = WellLogViewProps;
export const argTypesWellLogViewScrollerProp = argTypesWellLogViewProp;

const WellLogViewWithScroller: React.FC<WellLogViewWithScrollerProps> = (
    props
) => {
    const controllerRef = useRef<WellLogController | null>(null);
    const scrollerRef = useRef<ScrollerRef>(null);
    const skipScrollNotificationRef = useRef<number>(0);

    const calcPosTrack = useCallback((f: number): number => {
        const controller = controllerRef.current;
        if (!controller) return 0;
        const posMax = controller.getTrackScrollPosMax();
        const posTrack = f * posMax;
        return Math.round(posTrack);
    }, []);

    const getContentPosFraction = useCallback((): number => {
        const controller = controllerRef.current;
        if (!controller) return 0;
        const baseDomain = controller.getContentBaseDomain();
        const domain = controller.getContentDomain();
        const w = baseDomain[1] - baseDomain[0] - (domain[1] - domain[0]);
        return w ? (domain[0] - baseDomain[0]) / w : 0;
    }, []);

    const getTrackPosFraction = useCallback((): number => {
        const controller = controllerRef.current;
        if (!controller) return 0;
        return controller.getTrackScrollPosMax()
            ? controller.getTrackScrollPos() / controller.getTrackScrollPosMax()
            : 0.0;
    }, []);

    const setScrollerPosAndZoom = useCallback((): void => {
        let x: number, y: number;
        let xZoom: number, yZoom: number;
        const scroller = scrollerRef.current;
        if (!scroller) return;
        const controller = controllerRef.current;
        if (!controller) {
            x = y = 0.0;
            xZoom = yZoom = 1.0;
        } else {
            const contentZoom = controller.getContentZoom();
            const trackZoom = controller.getTrackZoom();
            xZoom = props.horizontal ? contentZoom : trackZoom;
            yZoom = props.horizontal ? trackZoom : contentZoom;

            const fContent = getContentPosFraction();
            const fTrack = getTrackPosFraction();
            x = props.horizontal ? fContent : fTrack;
            y = props.horizontal ? fTrack : fContent;
        }

        scroller.zoom(xZoom, yZoom);

        let shouldUpdateScroller = 2;
        {
            // compare with current values from scroller
            const _x = scroller.getScrollX();
            const _y = scroller.getScrollY();
            const _posTrack = calcPosTrack(props.horizontal ? _y : _x);
            const posTrack = calcPosTrack(props.horizontal ? y : x);
            if (posTrack === _posTrack) {
                shouldUpdateScroller--;
                if (props.horizontal) {
                    y = _y;
                } else {
                    x = _x;
                }
            }
            const _fContent = props.horizontal ? _x : _y;
            const fContent = props.horizontal ? x : y;
            if (Math.abs(fContent - _fContent) < 0.001) {
                shouldUpdateScroller--;
                if (props.horizontal) {
                    x = _x;
                } else {
                    y = _y;
                }
            }
        }
        if (shouldUpdateScroller) {
            if (scroller.scrollTo(x, y)) skipScrollNotificationRef.current++; // skip self-induced notification
        }
    }, [
        props.horizontal,
        getContentPosFraction,
        getTrackPosFraction,
        calcPosTrack,
    ]);

    // callback function from WellLogView
    const onCreateController = useCallback(
        (controller: WellLogController) => {
            controllerRef.current = controller;
            props.onCreateController?.(controller);
        },
        [props]
    );

    // callback function from WellLogView
    const onTrackScroll = useCallback(() => {
        setScrollerPosAndZoom();
        props.onTrackScroll?.();
    }, [setScrollerPosAndZoom, props]);

    // callback function from WellLogView
    const onTrackSelection = useCallback(() => {
        props.onTrackSelection?.();
    }, [props]);

    // callback function from WellLogView
    const onContentRescale = useCallback(() => {
        setScrollerPosAndZoom();
        props.onContentRescale?.();
    }, [setScrollerPosAndZoom, props]);

    // callback function from WellLogView
    const onContentSelection = useCallback(() => {
        props.onContentSelection?.();
    }, [props]);

    // callback function from Scroller
    const onScrollerScroll = useCallback(
        (x: number, y: number) => {
            if (skipScrollNotificationRef.current) {
                // the notification is self-induced
                skipScrollNotificationRef.current--;
                return;
            }

            const controller = controllerRef.current;
            if (!controller) return;

            const fContent = props.horizontal ? x : y; // fraction
            controller.scrollContentTo(fContent);

            const posTrack = calcPosTrack(props.horizontal ? y : x);
            controller.scrollTrackTo(posTrack);
        },
        [props.horizontal, calcPosTrack]
    );

    useEffect(() => {
        setScrollerPosAndZoom();
    }, [setScrollerPosAndZoom]); // componentDidMount equivalent

    return (
        <Scroller ref={scrollerRef} onScroll={onScrollerScroll}>
            <WellLogView
                // copy all props
                {...props}
                // redefine some callbacks
                onCreateController={onCreateController}
                onTrackScroll={onTrackScroll}
                onTrackSelection={onTrackSelection}
                onContentRescale={onContentRescale}
                onContentSelection={onContentSelection}
            />
        </Scroller>
    );
};

// Add propTypes for backward compatibility
// eslint-disable-next-line @typescript-eslint/no-explicit-any
(WellLogViewWithScroller as any).propTypes = _propTypesWellLogView();

export default WellLogViewWithScroller;
