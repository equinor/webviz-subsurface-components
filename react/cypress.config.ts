import { defineConfig } from "cypress";
import getCompareSnapshotsPlugin from "cypress-image-diff-js/dist/plugin";

export default defineConfig({
  component: {
    devServer: {
      framework: "create-react-app",
      bundler: "webpack",
    },
    video:false,
    setupNodeEvents(on, config) {
      getCompareSnapshotsPlugin(on, config),
      on('before:browser:launch', (browser = {}, launchOptions) => {
        if (browser.family === 'chromium' && browser.name !== 'electron') {
          launchOptions.args.push('--start-fullscreen')
      
          return launchOptions
        }
        if (browser.name === 'electron') {
          launchOptions.preferences.fullscreen = true
      
          return launchOptions
        }
      })
    }
  },
});
