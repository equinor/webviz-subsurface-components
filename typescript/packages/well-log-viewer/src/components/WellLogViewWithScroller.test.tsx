import { colorTables } from "@emerson-eps/color-tables";
import { render } from "@testing-library/react";
import "jest-styled-components";
import React from "react";

import type { Template } from "./WellLogTemplateTypes";
import WellLogViewWithScroller from "./WellLogViewWithScroller";
import type { ColorMapFunction } from "../utils/color-function";

import wellLogJson from "../../../../../example-data/L898MUD.json";
import templateJson from "../../../../../example-data/welllog_template_1.json";

const exampleColorMapFunctions = colorTables as ColorMapFunction[];

window.ResizeObserver =
    window.ResizeObserver ||
    jest.fn().mockImplementation(() => ({
        disconnect: jest.fn(),
        observe: jest.fn(),
        unobserve: jest.fn(),
    }));

describe("Test Well Log View Component with Scroller", () => {
    it("snapshot test", () => {
        const { container } = render(
            <WellLogViewWithScroller
                wellLogSets={wellLogJson}
                template={templateJson as Template}
                colorMapFunctions={exampleColorMapFunctions}
                primaryAxis={"md"}
                axisTitles={{
                    md: "MD",
                    tvd: "TVD",
                    time: "TIME",
                }}
                axisMnemos={{
                    md: ["DEPTH", "DEPT", "MD", "TDEP", "MD_RKB"],
                    tvd: ["TVD", "TVDSS", "DVER", "TVD_MSL"],
                    time: ["TIME"],
                }}
                options={{
                    checkDatafileSchema: true,
                    hideTrackLegend: true,
                    hideTrackTitle: true,
                }}
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
