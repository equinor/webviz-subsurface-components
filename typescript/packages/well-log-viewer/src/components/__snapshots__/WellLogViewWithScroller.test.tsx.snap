// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Well Log View Component with Scroller snapshot test 1`] = `
<div
  style="overflow: scroll; width: 100%; height: 100%;"
>
  <div
    style="width: 0px; height: 0px;"
  >
    <div
      style="position: absolute;"
    >
      <div
        class="welllogview"
        style="width: 100%; height: 100%; display: flex; flex-direction: column;"
      >
        <div
          style="width: 100%; height: 100%; display: flex; flex-direction: column;"
        >
          <div
            class="log-controller"
          >
            <div
              class="overlay"
              style="width: 0px; left: 0px; overflow: hidden;"
            >
              <div
                class="depth"
                style="position: absolute; pointer-events: none; visibility: hidden;"
              />
              <div
                class="rubber-band"
                style="position: relative; pointer-events: none; height: 9px; width: 100%; visibility: hidden;"
              >
                <div
                  style="height: 1px; width: 100%; top: 4.5px; position: relative;"
                />
              </div>
              <div
                class="pinned"
                style="position: absolute; pointer-events: none; height: 9px; width: 100%; left: 0px; visibility: hidden;"
              >
                <div
                  style="height: 1px; width: 100%; top: 4.5px; position: absolute;"
                />
              </div>
            </div>
            <div
              class="track"
              style="max-width: 50px; flex: 2;"
            >
              <div
                class="track-container"
              >
                <svg
                  style="position: absolute;"
                />
              </div>
            </div>
            <div
              class="track"
              style="max-width: 50px; flex: 2;"
            >
              <div
                class="track-container"
              >
                <svg
                  style="position: absolute;"
                />
              </div>
            </div>
            <div
              class="track"
              style="display: flex; flex: 3;"
            >
              <div
                class="track-container"
              >
                <canvas
                  style="position: absolute;"
                />
              </div>
            </div>
            <div
              class="track"
              style="display: flex; flex: 3;"
            >
              <div
                class="track-container"
              >
                <canvas
                  style="position: absolute;"
                />
              </div>
            </div>
            <div
              class="track"
              style="display: flex; flex: 3;"
            >
              <div
                class="track-container"
              >
                <canvas
                  style="position: absolute;"
                />
              </div>
            </div>
            <div
              class="track"
              style="display: flex; flex: 3;"
            >
              <div
                class="track-container"
              >
                <canvas
                  style="position: absolute;"
                />
              </div>
            </div>
            <div
              class="track"
              style="display: flex; flex: 3;"
            >
              <div
                class="track-container"
              >
                <canvas
                  style="position: absolute;"
                />
              </div>
            </div>
            <div
              class="track"
              style="display: none; flex: 3;"
            >
              <div
                class="track-container"
              >
                <canvas
                  style="position: absolute;"
                />
              </div>
            </div>
            <div
              class="track"
              style="display: none; flex: 3;"
            >
              <div
                class="track-container"
              >
                <canvas
                  style="position: absolute;"
                />
              </div>
            </div>
            <div
              class="track"
              style="display: none; flex: 3;"
            >
              <div
                class="track-container"
              >
                <canvas
                  style="position: absolute;"
                />
              </div>
            </div>
            <div
              class="track"
              style="display: none; flex: 3;"
            >
              <div
                class="track-container"
              >
                <canvas
                  style="position: absolute;"
                />
              </div>
            </div>
            <div
              class="track"
              style="display: none; flex: 3;"
            >
              <div
                class="track-container"
              >
                <canvas
                  style="position: absolute;"
                />
              </div>
            </div>
            <div
              class="track"
              style="display: none; flex: 3;"
            >
              <div
                class="track-container"
              >
                <canvas
                  style="position: absolute;"
                />
              </div>
            </div>
            <div
              class="track"
              style="display: none; flex: 3;"
            >
              <div
                class="track-container"
              >
                <canvas
                  style="position: absolute;"
                />
              </div>
            </div>
            <div
              class="track"
              style="display: none; flex: 3;"
            >
              <div
                class="track-container"
              >
                <canvas
                  style="position: absolute;"
                />
              </div>
            </div>
            <div
              class="track"
              style="display: none; flex: 3;"
            >
              <div
                class="track-container"
              >
                <canvas
                  style="position: absolute;"
                />
              </div>
            </div>
            <div
              class="track"
              style="display: none; flex: 3;"
            >
              <div
                class="track-container"
              >
                <canvas
                  style="position: absolute;"
                />
              </div>
            </div>
            <div
              class="track"
              style="display: none; flex: 3;"
            >
              <div
                class="track-container"
              >
                <canvas
                  style="position: absolute;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
